### PR TITLE
Investigate flaky metrics tests

### PR DESCRIPTION
**Pull request by rajivnathan**
Adds a wait for pending usersignup deletions before capturing baseline metrics.

### CODE REVIEW

File: wait/host.go [hunk 1]
```diff
@@ -254,6 +254,26 @@ func UntilUserSignupHasStateLabel(expLabelValue string) UserSignupWaitCriterion
...
+	time.Sleep(initialDelay)
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		usList := &toolchainv1alpha1.UserSignupList{}
+		if err := a.Client.List(context.TODO(), usList, client.InNamespace(a.Namespace)); err != nil {
+			return false, err
+		}
+		for _, us := range usList.Items {
+			if us.DeletionTimestamp != nil {
+				a.T.Logf("a UserSignup is being deleted: %s", us.Name)
+				return false, err
```
Comment (alexeykazakov) [inline wait/host.go:268]: Just to improve code readability a bit.
```suggestion
				return false, nil
```

**Review by alexeykazakov — APPROVED** [review]
Looks good for now. But we need to think about how we could refactor the tests so they do not depend on resources are not directly involved in the test. It could be tricky to isolate the metric tests though.
It's not urgent though.

**Comment by rajivnathan** [comment]
Looks like it failed due to an error not related to metrics:
```
    host.go:186: waiting for status condition of MasterUserRecord 'johnsmith'. Actual: '[{Type:Ready Status:False LastTransitionTime:2020-11-21 16:52:27 +0000 UTC Reason:Provisioning Message: LastUpdatedTime:<nil>}]'; Expected: '[{Type:Ready Status:True LastTransitionTime:0001-01-01 00:00:00 +0000 UTC Reason:Provisioned Message: LastUpdatedTime:<nil>} {Type:UserProvisionedNotificationCreated Status:True LastTransitionTime:0001-01-01 00:00:00 +0000 UTC Reason:NotificationCRCreated Message: LastUpdatedTime:<nil>}]'
    user_setup.go:77: 
        	Error Trace:	user_setup.go:77
        	            				e2e_test.go:66
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestE2EFlow
    client.go:83: resource type  with namespace/name (codeready-toolchain-host-21164251/9fa57c0a-1894-4b42-a51c-67984b873f9e) successfully deleted
    client.go:83: resource type  with namespace/name (codeready-toolchain-host-21164251/a4c4a37a-13fc-4e01-9b07-3f97bf43e249) successfully deleted
    client.go:83: resource type  with namespace/name (codeready-toolchain-host-21164251/69fb40e9-7435-4b36-a47b-5524ceaafcd4) successfully deleted
    client.go:83: resource type  with namespace/name (codeready-toolchain-host-21164251/1111b0e9-66d7-4eb3-9efc-3b5eae0c4b6f) successfully deleted
    client.go:83: resource type  with namespace/name (codeready-toolchain-host-21164251/29dc568d-4548-4c8e-ae61-cdf9464d1d2f) successfully deleted
--- FAIL: TestE2EFlow (72.14s)
    --- PASS: TestE2EFlow/verify_cluster_statuses_are_valid (0.33s)
        --- PASS: TestE2EFlow/verify_cluster_statuses_are_valid/verify_member_cluster_status (0.17s)
        --- PASS: TestE2EFlow/verify_cluster_statuses_are_valid/verify_overall_toolchain_status (0.17s)
    --- PASS: TestE2EFlow/verify_metrics_servers (0.33s)
        --- PASS: TestE2EFlow/verify_metrics_servers/verify_host_metrics_server (0.16s)
        --- PASS: TestE2EFlow/verify_metrics_servers/verify_member_metrics_server (0.17s)
FAIL
FAIL	github.com/codeready-toolchain/toolchain-e2e/test/e2e	72.172s
FAIL
time="2020-11-21T16:53:28Z" level=fatal msg="Failed to build test binary: failed to exec []string{\"go\", \"test\", \"./test/e2e/...\", \"-namespacedMan\", \"deploy/test/empty.yaml\", \"-globalMan\", \"deploy/test/empty.yaml\", \"-root\", \"/go/src/github.com/codeready-toolchain/toolchain-e2e\", \"-test.timeout=60m\", \"-test.failfast\", \"-parallel=1\", \"-skipCleanupOnError=false\"}: exit status 1"
```

**Comment by rajivnathan** [comment]
CI error:
`error: unable to extract layer sha256:72f24a12593c2c5f14d1b89ea52905aa0f116b215cc5c06f1a9f16f2d9979257 from registry.build01.ci.openshift.org/ci-op-hvkx2365/stable@sha256:d374ab4a9324381f06d10d664c9ce811c7ed2518dbdf1de4216d5270cc921f56: received unexpected HTTP status: 503 Service Unavailable`

**Pull request merged**